### ISSUE TITLE
Issue when wiping buffers

### ISSUE THREAD

**Issue by pavkam** [Wansmer/symbol-usage.nvim#53]
I written my own session management code that, on restore, deletes and wipes all buffers and then loads the new session from the file. When that happens I get the following error:
```
Error executing vim.schedule lua callback: ...e/nvim/lazy/symbol-usage.nvim/lua/symbol-usage/langs.lua:26: Invalid buffer id: 90
stack traceback:
	[C]: in function 'nvim_buf_get_lines'
	...e/nvim/lazy/symbol-usage.nvim/lua/symbol-usage/langs.lua:26: in function 'cb'
	...e/nvim/lazy/symbol-usage.nvim/lua/symbol-usage/utils.lua:62: in function 'every'
	.../nvim/lazy/symbol-usage.nvim/lua/symbol-usage/worker.lua:118: in function 'is_need_count'
	.../nvim/lazy/symbol-usage.nvim/lua/symbol-usage/worker.lua:164: in function '_walk'
	.../nvim/lazy/symbol-usage.nvim/lua/symbol-usage/worker.lua:199: in function '_walk'
	.../nvim/lazy/symbol-usage.nvim/lua/symbol-usage/worker.lua:222: in function 'traversal'
	.../nvim/lazy/symbol-usage.nvim/lua/symbol-usage/worker.lua:81: in function 'handler'
	...w/Cellar/neovim/0.9.5/share/nvim/runtime/lua/vim/lsp.lua:1393: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
```

See my config: https://github.com/pavkam/dotfiles/blob/1514359b9ff58c08f3d525b76a026a6443dc9eb2/.config/nvim/lua/core/session.lua#L51

Them buffers are gone at some point due to my code wiping them and loading others instead. I think there needs to be code that checks that a buffer is valid before doing any other work.

Thanks for the great project,
Alex.

**Comment by pavkam** [comment]
Managed to get around the error by modifying my function as follows:
```
function M.restore_session(name)
    if utils.has_plugin 'symbol-usage.nvim' then -- HACK: disable symbol-usage before restoring session
        pcall((require 'symbol-usage').toggle_globally)
    end

    vim.defer_fn(function()
        -- close all windows, tabs and buffers
        vim.cmd [[silent! tabonly!]]
        vim.cmd [[silent! %bd!]]
        vim.cmd [[silent! %bw!]]

        -- restore session and shada files
        local session_file_name = name .. '.vim'
        local shada_file_name = name .. '.shada'

        if vim.fn.filereadable(session_file_name) == 1 then
            vim.cmd('source ' .. session_file_name)
        end
        if vim.fn.filereadable(shada_file_name) == 1 then
            vim.cmd('rshada ' .. shada_file_name)
        end

        if utils.has_plugin 'symbol-usage.nvim' then
            pcall((require 'symbol-usage').toggle_globally)
        end
    end, 0)
end

```

**Comment by Wansmer** [comment]
Thanks for the report. I fixed it in #54.
